packages/core/src/middleware/koa-log.ts

```diff
@@ -10,17 +10,52 @@ export type WithLogContext<ContextT> = ContextT & {
   log: <T extends LogType>(type: T, payload: LogPayloads[T]) => void;
 };
 
-const saveLog = async (type: LogType, payload: LogPayload) => {
-  try {
-    await insertLog({
-      id: nanoid(),
-      type,
-      payload,
-    });
-  } catch (error: unknown) {
-    console.error('An error occurred while inserting log');
-    console.error(error);
-  }
+const initLog = () => {
+  // eslint-disable-next-line @silverhand/fp/no-let
+  let type: Optional<LogType>;
+  // eslint-disable-next-line @silverhand/fp/no-let
+  let payload: LogPayload = {};
+
+  const mergeLog = (logType: LogType, logPayload: LogPayload) => {
+    if (type !== logType) {
+      // eslint-disable-next-line @silverhand/fp/no-mutation
+      payload = {}; // Reset payload when type changes
+    }
+
+    // eslint-disable-next-line @silverhand/fp/no-mutation
+    type = logType; // Use first initialized log type
+    // eslint-disable-next-line @silverhand/fp/no-mutation
+    payload = deepmerge(payload, logPayload);
+  };
+
+  const mergeLogError = (error: unknown) => {
+    if (type) {
```

Comment:
it doesn't matter if we have `type` or not here?

Reply:
updated.